### ISSUE TITLE
New versioning scheme

### ISSUE THREAD

**Issue by madalinignisca** [hetznercloud/hcloud-cloud-controller-manager#672]
### TL;DR

Change the versioning scheme to reflect Kubernetes releases. This would allow people to match a release that ensures compatibility with the used Kubernetes version without figuring it out from a table. Tags could reference the same commit with the intention of helping a person match the compatible Kubernetes release they require.

Although a release could match multiple compatible Kubernetes versions, multiple tags/releases in GitHub should not create issues and could probably also be automated.

### Expected behavior

If I have Kubernetes 1.28.k, I would be able to install Hetzner CCM 1.28.h, which has been ensured to be compatible with my version of Kubernetes.

The .h would be incremental numbering of release for Hetzner CCM, and nothing to do with the .k in kubernetes release, as the target is on Major.Minor, and not Patch.

**Comment by apricote** [comment]
Thanks for the suggestion @madalinignisca,

I am personally against tying our versions to Kubernetes. Let me explain my reasons:

## Kubernetes Version Compatibility

The matrix of supported versions we have in the README only shows the versions we test against. This does not mean that older/newer versions of Kubernetes do not work with the HCCM version.

The container itself should work all the way back to ~1.20 at least, there might be some some new resources or fields that we use in our deployments manifests.

## Feature Availability

We want to make new features and bugfixes available to customers as quickly as possible. If we had separate releases for every Kubernetes version, we would need to maintain 3-4 branches at a time and backport features+fixes to them. This is too much effort for our little team.

## Docs

I think we could do better in highlighting the supported Kubernetes versions in the Release Notes for every release, then users have an easier time finding this info.

We could also clean up the README and separate the manifest-yaml links from the HCCM<>Kubernetes version matrix.

**Comment by madalinignisca** [comment]
I was not thinking of using multiple branches for different majors and minors, but tagging/releasing from the main as it is now and creating all corresponding tags/releases. This would not involve any extra branches.

If the releases cover all current supported Kubernetes versions, I do not see the reason to backport patches to older unsupported versions.

Currently, the CCM looks good on all current supported Kubernetes versions; I'm not very sure about the latest, which can always be suggested to deploy from the main.

Similar CRI-O follows this pattern, and I think many other Kubernetes add-ons will follow up.

But, it's just my opinion and was a suggestion. I'm happy also with generic semantic versioning.
